pkg/controller/nodenetworkconfigurationpolicy/selectors/manager_test.go

```diff
@@ -0,0 +1,140 @@
+package selectors
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	nmstatev1alpha1 "github.com/nmstate/kubernetes-nmstate/pkg/apis/nmstate/v1alpha1"
+)
+
+var _ = Describe("NodeNetworkConfigurationPolicy controller selectors", func() {
+	var ExpectedNode = "node01"
+	type nodeSelectorCase struct {
+		ObjNodeName     string
+		NodeSelector    map[string]string
+		NodeLabels      map[string]string
+		Matches         bool
+		UnmatchedLabels map[string]string
+	}
+	DescribeTable("testing node selectors",
+		func(c nodeSelectorCase) {
+			node := corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:   c.ObjNodeName,
+					Labels: c.NodeLabels,
+				},
+			}
+
+			policy := nmstatev1alpha1.NodeNetworkConfigurationPolicy{
+				Spec: nmstatev1alpha1.NodeNetworkConfigurationPolicySpec{
+					NodeSelector: c.NodeSelector,
+					DesiredState: nmstatev1alpha1.State{Raw: nil}, // TODO
```

Comment:
What is the TODO action?

Reply:
To be honest I don't remember, maybe it's related to advanced policy matchers, let's just remove DesiredState altogether since it's not needed for this.